packages/migrate/index.ts

```diff
@@ -74,129 +95,144 @@ module.exports = function migrate(...args) {
  * user decides to abort the migration
  */
 
-function runMigration(currentConfigPath, outputConfigPath) {
-	const recastOptions = {
-		quote: "single"
+function runMigration(currentConfigPath: string, outputConfigPath: string): Promise<void> | void {
+	const recastOptions: object = {
+		quote: "single",
 	};
-	const tasks = new Listr([
+
+	const tasks: Listr = new Listr([
 		{
-			title: "Reading webpack config",
-			task: ctx =>
-				new PLazy((resolve, reject) => {
+			task: (ctx: INode): string | void | Listr | Promise<{}> =>
+				new pLazy((
+						resolve: (value?: object) => void,
+						reject: (reason?: string | object, err?: object) => void,
+					) => {
 					fs.readFile(currentConfigPath, "utf8", (err, content) => {
 						if (err) {
 							reject(err);
 						}
 						try {
-							const jscodeshift = require("jscodeshift");
+							const jscodeshift: IJSCodeshift = require("jscodeshift");
 							ctx.source = content;
 							ctx.ast = jscodeshift(content);
 							resolve();
 						} catch (err) {
```

Comment:
missing types

Reply:
TS doesn't allow adding types in `catch` block of _try()_ _catch()_.
```
[ts] Catch clause variable cannot have a type annotation.
```

We're only adding error object in `catch` block of promises.

Reply:
weird. Okay